src/sc_keeper/rate_limit.py

```
@@ -0,0 +1,271 @@
+
+    def __init__(self, credits_per_minute: int):
+        self.credits_per_minute = credits_per_minute
+        # credit consumption history: {key: [(timestamp, credits), ...]}
+        self.windows: dict[str, list[tuple[float, int]]] = defaultdict(list)
+
+    def is_allowed(
+        self,
+        key: str,
+        credits_per_minute: Optional[int] = None,
+        credit_cost: int = 1,
+        **kwargs,
```

Ruff (0.14.10)

Unused method argument: `kwargs`

(ARG002)